backend/mobilityAI/Controllers/DevicesController.cs

```diff
@@ -68,19 +68,52 @@ public JsonResult GetDevice(string deviceId)
                                 [FromQuery]int patientId, 
                                 [FromQuery]string lastsync)
         {
+            //Check if valid patient ID entered
+            Patient p = (from a in _context.Patients
+                         where a.Id == patientId
+                         select a).SingleOrDefault();
+
+            if(p == null) {
+                return BadRequest("Patient does not exist");
+            }
+
             Device data = (from a in _context.Devices
                            where (a.Id == deviceId)
                            select a).SingleOrDefault();
 
-            var date = DateTime.Parse(lastsync);
+            if(data == null) {
+                //Add new device
+                Device nDev = new Device()
+                {
+                    Id = deviceId,
+                    FriendlyName = "MetaMotion",
+                    PatientID = patientId,
+                    LastSync = DateTime.Now
+                };
+
+                _context.Devices.Add(nDev);
+                _context.SaveChanges(); //save changes here to get the new ID
+
+                //Update patient
+                p.DeviceId = nDev.Id;
+            } else {
+                p.DeviceId = data.Id;
+
+                var date = new DateTime();
+
+                if(DateTime.TryParse(lastsync, out date)) {
+                    data.LastSync = date;
+                } else {
+                    data.LastSync = DateTime.Now;
+                }
 
-            data.Id = deviceId;
-            data.FriendlyName = name;
-            data.PatientID = patientId;
-            data.LastSync = date;
+                data.Id = deviceId;
+                data.FriendlyName = name;
+                data.PatientID = patientId;
+            }
 
             _context.SaveChanges();
-            return Ok();
+            return Ok(p.FirstName + " " + p.LastName);
```

Comment:
Is there a reason you're returning this?

Reply:
This is what's going to be displayed once the update is made (ie: the header bar on the android app will change from the old first name/last name to this.

Reply:
Oh ok, you should return the whole Patients object then and do the string concatenation on the android side.